numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2077,18 +2072,18 @@ def __cuda_stream__(self):
 
     def __repr__(self):
         default_streams = {
-            drvapi.CU_STREAM_DEFAULT: "<Default CUDA stream on %s>",
-            drvapi.CU_STREAM_LEGACY: "<Legacy default CUDA stream on %s>",
-            drvapi.CU_STREAM_PER_THREAD: "<Per-thread default CUDA stream on %s>",
+            drvapi.CU_STREAM_DEFAULT: "<Default CUDA stream>",
+            drvapi.CU_STREAM_LEGACY: "<Legacy default CUDA stream>",
+            drvapi.CU_STREAM_PER_THREAD: "<Per-thread default CUDA stream>",
         }
         ptr = self.handle.value or drvapi.CU_STREAM_DEFAULT
 
         if ptr in default_streams:
-            return default_streams[ptr] % self.context
+            return default_streams[ptr]
         elif self.external:
-            return "<External CUDA stream %d on %s>" % (ptr, self.context)
+            return f"<External CUDA stream {ptr:d}>"
         else:
-            return "<CUDA stream %d on %s>" % (ptr, self.context)
```

Comment:
The context was only being used for repring. The complexity of weak references doesn't seem justifiable for pretty printing, hence this PR :)